tests/components/tplink/test_common.py

```diff
@@ -0,0 +1,85 @@
+"""Common code tests."""
+from datetime import timedelta
+from unittest.mock import MagicMock
+
+from pyHS100 import SmartDeviceException
+
+from homeassistant.components.tplink.common import async_add_entities_retry
+from homeassistant.helpers.typing import HomeAssistantType
+
+
+async def test_async_add_entities_retry(
+        hass: HomeAssistantType
+):
+    """Test interval callback."""
+    async_add_entities_callback = MagicMock()
+
+    callback_side_effects = [
```

Comment:
Would it be possible to have some sort of explanation for what exactly is tested and where these side effect values come from? Just for the sake of saving some time for someone who's going to update the tests for any reason in the future.